universal-login-contracts/src/integration/ContractService.ts

```diff
@@ -0,0 +1,62 @@
+import {utils, providers} from 'ethers';
+import {computeCounterfactualAddress, createKeyPair, WALLET_MASTER_VERSIONS, ensureNotFalsy} from '@unilogin/commons';
+import {computeGnosisCounterfactualAddress} from '../gnosis-safe@1.1.1/utils';
+import {DEPLOY_CONTRACT_NONCE} from '../gnosis-safe@1.1.1/constants';
+import {ProviderService} from './ProviderService';
+
+export class ContractService {
```

Comment:
Functions typical for contract walle: createFutureWallet,  getInitCode, createFutureGnosis, fetchWalletVersion, fetchMasterAddress, fetchProxyVersion.
Others are typical for the provider.